### ISSUE TITLE
Exception when running script ex2_curved_slicing_advanced.py

### ISSUE THREAD

**Issue by yvanblanchard** [compas-dev/compas_slicer#156]
Hello,

when runnning the ex2_curved_slicing_advanced.py script, I get errors at the topological sorting step. I did not changed the original script and input files.

Thank you

![image](https://user-images.githubusercontent.com/63931618/155124805-74785a14-c2e0-40a1-b089-42ebabe54547.png)

**Comment by ioannaMitropoulou** [comment]
Hello! Could you make sure you have pulled the latest version of compas_slicer? 
I just ran this example, it seems to work. The resulting paths visualized using the grasshopper component 'csLoadSlicer' are the following:
![Untitled-1](https://user-images.githubusercontent.com/17250275/155358898-4ad55685-42d6-45f8-9342-4f74a034c7d6.png)
If after updating to the latest compas_slicer you still run into this problem, could you zip the entire example folder and send it to me to see if I can reproduce it? 

Thanks,
Ioanna

**Comment by yvanblanchard** [comment]
Hello,
I pulled the latest state of code, and also the new example python script (the latest one uses 'implify_paths_rdp_igl' with IGL).
But I finally have the same issue (assertion error when doing topological sorting).
Here is my example folder , containing the resulting files, zipped.
[data_advanced_example-ybd.zip](https://github.com/compas-dev/compas_slicer/files/8127541/data_advanced_example-ybd.zip)

**Comment by yvanblanchard** [comment]
Hello, Waiting for an explanation (I hope!) using my provided data set, could it be possible to get the json outpoints files for this shape please ?
thank you very much for your help.

**Comment by ioannaMitropoulou** [comment]
Hello,
I haven't managed to reproduce the error. When I ran the example on the files you sent me, I get the expected results. I send you back the folder with the data you sent me, the python script I used for the slicing (just a copy of the example2), all the resulting jsons, and the grasshopper file for visualizing the results.

![image](https://user-images.githubusercontent.com/17250275/155671806-0ed204a8-7b2e-48f5-a678-1278a39bd907.png)

[example-ybd.zip](https://github.com/compas-dev/compas_slicer/files/8139266/example-ybd.zip)
